### PR TITLE
Refactoring/tao 5102/remove globals from core extension

### PR DESCRIPTION
**Pull request by ionutpad**

### CODE REVIEW

File: manifest.php [hunk 1]
```diff
@@ -25,7 +25,7 @@
     'label' => 'Linear Test Model',
     'description' => 'A simple linear test model',
     'license' => 'GPL-2.0',
-    'version' => '3.0.0',
+    'version' => '3.1.0',
     'author' => 'Open Assessment Technologies SA',
     'requires' => array(
         'tao' => '>=9.0.0',
```
Comment (zagovorichev) [inline manifest.php]: tao => '12.21.0' ?

File: model/storage/DeprecatedStorage.php [hunk 2]
```diff
@@ -85,15 +86,15 @@ public function load(core_kernel_classes_Resource $test)
...
+        $directoryId = $test->getOnePropertyValue($this->getProperty(TestService::TEST_TESTCONTENT_PROP));
         if(is_null($directoryId)){
             throw new \common_exception_FileSystemError(__('Unknown test directory'));
         }
 
         $directory = \tao_models_classes_service_FileStorage::singleton()->getDirectoryById($directoryId->literal);
         $directory->deleteSelf();
 
-		$test->removePropertyValues($this->getProperty(TEST_TESTCONTENT_PROP));
+		$test->removePropertyValues($this->getProperty(TestService::TEST_TESTCONTENT_PROP));
```
Comment (zagovorichev) [inline model/storage/DeprecatedStorage.php]: it is deprecated, why do you use it?

File: test/model/TestModelTest.php [hunk 3]
```diff
@@ -233,7 +233,7 @@ public function testGetConfig(){
             ->disableOriginalConstructor()
             ->setMethods(array('getOnePropertyValue'))
             ->getMock();
-        $propInstanceContent = new \core_kernel_classes_Property(TEST_TESTCONTENT_PROP);
+        $propInstanceContent = new \core_kernel_classes_Property(TestService::TEST_TESTCONTENT_PROP);
```
Comment (zagovorichev) [inline test/model/TestModelTest.php]: all TEST_TESTCONTENT_PROP are deprecated, replace it, please

**Review by zagovorichev — CHANGES_REQUESTED** [review]

File: model/storage/DeprecatedStorage.php (hunk 2 above)
Comment (ionutpad) [reply model/storage/DeprecatedStorage.php]: It's just a replace this entire class it's deprecated. So I am not using it just replace it.

File: test/model/TestModelTest.php (hunk 3 above)
Comment (ionutpad) [reply test/model/TestModelTest.php]: Same case here, just replace not really usages

File: model/storage/DeprecatedStorage.php (hunk 2 above)
Comment (zagovorichev) [reply model/storage/DeprecatedStorage.php]: what do I mean:
https://github.com/oat-sa/extension-tao-test/blob/476db8e24a01aee6765da5de09da4908ceaeb255/models/classes/class.TestsService.php#L46
it's deprecated, so you can just use
TestService::PROPERTY_TEST_CONTENT

Comment (ionutpad) [reply model/storage/DeprecatedStorage.php]: I see now, I have replace all of TEST_TESTCONTENT_PROP with  PROPERTY_TEST_CONTENT, Thanks

File: manifest.php [hunk 4]
```diff
@@ -25,10 +25,10 @@
     'label' => 'Linear Test Model',
     'description' => 'A simple linear test model',
     'license' => 'GPL-2.0',
-    'version' => '3.0.0',
+    'version' => '3.1.0',
     'author' => 'Open Assessment Technologies SA',
     'requires' => array(
-        'tao' => '>=9.0.0',
+        'tao' => '>=14.3.1',
```
Comment (zagovorichev) [inline manifest.php]: why 14.3.1? new feature was created in 12.21.0

**Review by zagovorichev — CHANGES_REQUESTED** [review]

File: manifest.php (hunk 4 above)
Comment (ionutpad) [reply manifest.php]: I checked some when was merged and it was 14.3.1 but I was wrong it's actually 12.21.4.
https://github.com/oat-sa/tao-core/pull/1477

**Review by zagovorichev — APPROVED** [review]

**Pull request merged**